pkg/operator/podsecurityreadinesscontroller/violation.go

```diff
@@ -0,0 +1,130 @@
+package podsecurityreadinesscontroller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	applyconfiguration "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/klog/v2"
+	psapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	syncerControllerName = "pod-security-admission-label-synchronization-controller"
+)
+
+var (
+	alertLabels = sets.Set[string]{
+		psapi.WarnLevelLabel:  {},
+		psapi.AuditLevelLabel: {},
+	}
+)
+
+func (c *PodSecurityReadinessController) isNamespaceViolating(ctx context.Context, ns *corev1.Namespace) (bool, error) {
+	viableLabels := getSyncerLabels(ns.ManagedFields).Intersection(alertLabels)
+	if viableLabels.Len() == 0 {
+		// If there are no labels managed by the syncer, we can't make a decision.
+		return false, nil
```

Comment:
This one is tricky, but I would not to prefer to assume that the customer set reasonable values.